src/main/java/randoop/generation/Impurity.java

```diff
@@ -0,0 +1,518 @@
+    } else if (outputClass == float.class) {
+      return (float) randomGaussian;
+    } else if (outputClass == double.class) {
+      return randomGaussian;
+    } else {
+      throw new RuntimeException("Unexpected primitive type: " + outputClass.getName());
+    }
+  }
+
+  /**
+   * Get the method (in a list) that can be used to fuzz primitive numbers.
+   *
+   * @param outputClass the class of the primitive number to be fuzzed
+   * @return a list of methods that can be used to fuzz primitive numbers
+   * @throws NoSuchMethodException if no suitable method is found for the given class
+   */
+  private static List<Method> getNumberFuzzingMethod(Class<?> outputClass)
+      throws NoSuchMethodException {
+
+    List<Method> methodList = new ArrayList<>();
+
+    // Map each wrapper to its primitive type and a common method
+    if (outputClass == int.class) {
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+    } else if (outputClass == double.class) {
+      methodList.add(Double.class.getMethod("sum", double.class, double.class));
+    } else if (outputClass == float.class) {
+      methodList.add(Float.class.getMethod("sum", float.class, float.class));
+    } else if (outputClass == long.class) {
+      methodList.add(Long.class.getMethod("sum", long.class, long.class));
+    } else if (outputClass == short.class) {
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+    } else if (outputClass == byte.class) {
+      throw new NoSuchMethodException("Byte fuzzing is not supported yet");
+    } else {
+      throw new NoSuchMethodException("Object fuzzing is not supported");
+    }
+
+    if (methodList.isEmpty()) {
+      // Should be unreachable
+      throw new NoSuchMethodException(
+          "Unable to find suitable method for class: "
+              + outputClass.getName()
+              + " in primitive number fuzzing");
+    }
+
+    return methodList;
+  }
+
+  /**
+   * Get a sequence with the fuzzing statement appended at the end for fuzzing a String.
+   *
+   * @param sequence the sequence to append the fuzzing sequence to
+   * @param fuzzingOperationIndex the index of the fuzzing operation to perform
+   * @param fuzzStatementOffset the offset counter for the number of fuzzing statements added
+   * @return a sequence with the fuzzing statement appended at the end
+   * @throws IllegalArgumentException if the fuzzing operation is invalid
+   * @throws IndexOutOfBoundsException if the input String is empty
+   */
+  private static Sequence getFuzzedSequenceForString(
+      Sequence sequence, int fuzzingOperationIndex, FuzzStatementOffset fuzzStatementOffset)
+      throws IllegalArgumentException, IndexOutOfBoundsException {
+    sequence = initializeWithStringBuilder(sequence, fuzzStatementOffset);
+
+    Object stringValue = getStringValue(sequence);
+    int stringLength = stringValue.toString().length();
+
+    if (stringLength == 0) {
```

Comment:
Since this is a precondition for callers, it should be stated in the method documentation.